src/conf.c

```diff
@@ -730,6 +742,16 @@ bool config_write(meshlink_handle_t *mesh, const char *conf_subdir, const char *
 		return false;
 	}
 
+	// Sync the renamed file's directory where the file system looks to find your file
+	// and to avoid system failure (such as power loss, ENOSPC or an I/O error)
+	snprintf(path, sizeof(path), "%s" SLASH "%s" SLASH "hosts", mesh->confbase, conf_subdir);
+
+	if(!sync_path(path)) {
```

Comment:
Syncing directories is not about preventing corruption, it is to ensure that if a crash happens right after the sync, we are sure that we see the new name after we restart. In many cases we don't need this guarantee; I mean we could have a power failure right before `sync_path()` is called, and that should be fine too.